Philips.CodeAnalysis.SecurityAnalyzers/LicenseAnalyzer.cs

```diff
@@ -1,43 +1,733 @@
+				ReportDebugDiagnostic(context, $"Package {package.Name} {package.Version ?? "unknown"}: License = {displayLicense}");
+
+				// Check if license is acceptable
+				// Temporarily remove condition to test if PH2155 diagnostics are created
+				//if (!string.IsNullOrEmpty(licenseInfo) && !IsLicenseAcceptable(licenseInfo, allowedLicenses))
+				{
+					var diagnostic = Diagnostic.Create(
+						Rule,
+						Location.None,
+						package.Name,
+						package.Version ?? "unknown",
+						licenseInfo);
+
+					context.ReportDiagnostic(diagnostic);
+				}
+			}
+
+			// Save updated license cache
+			SaveLicenseCache(licenseCache, Path.GetDirectoryName(assetsFilePath));
+		}
+
+		private static string TryFindAssetsFileFromSourcePaths(CompilationAnalysisContext context)
+		{
+			// Search for project.assets.json using actual source file paths from the compilation
+			// instead of Directory.GetCurrentDirectory() which points to VS install directory
+
+			var sourceDirectories = context.Compilation.SyntaxTrees
+				.Where(tree => !string.IsNullOrEmpty(tree.FilePath))
+				.Select(tree => Path.GetDirectoryName(tree.FilePath))
+				.Where(dir => !string.IsNullOrEmpty(dir))
+				.Distinct(StringComparer.OrdinalIgnoreCase)
+				.ToList();
+
+			// Search in each source directory and its parent directories
+			foreach (var sourceDir in sourceDirectories)
+			{
+				var foundPath = SearchForAssetsFile(context, sourceDir, enableDetailedLogging: false);
+				if (foundPath != null)
+				{
+					return foundPath;
+				}
+			}
+
+			// Fallback to current directory search if source paths didn't work
+			var currentDir = Directory.GetCurrentDirectory();
+			var fallbackPath = SearchForAssetsFile(context, currentDir, enableDetailedLogging: false);
+			if (fallbackPath != null)
+			{
+				return fallbackPath;
+			}
+
+			// File not found - now show detailed diagnostics to help troubleshoot
+			ReportDebugDiagnostic(context, $"Could not find {ProjectAssetsFileName}. Starting detailed search diagnostics...");
+			ReportDebugDiagnostic(context, $"Found {sourceDirectories.Count} unique source directories from compilation");
+
+			// Re-search with detailed logging enabled
+			foreach (var sourceDir in sourceDirectories)
+			{
+				ReportDebugDiagnostic(context, $"Searching from source directory: {sourceDir}");
+				_ = SearchForAssetsFile(context, sourceDir, enableDetailedLogging: true);
+			}
+
+			// Fallback search with detailed logging
+			ReportDebugDiagnostic(context, $"Source directory search failed, falling back to current directory: {currentDir}");
+			_ = SearchForAssetsFile(context, currentDir, enableDetailedLogging: true);
+
+			ReportDebugDiagnostic(context, $"Exhaustive search completed, {ProjectAssetsFileName} not found");
+			return null;
+		}
+
+		private static string SearchForAssetsFile(CompilationAnalysisContext context, string startDirectory, bool enableDetailedLogging)
```

Comment:
From Sonarcloud: Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed.